hcloud/data_source_hcloud_sshkeys_test.go

```diff
@@ -0,0 +1,59 @@
+package hcloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+const dataSSHKeysConfig = `
+data "hcloud_ssh_keys" "all_keys" {
+}
+data "hcloud_ssh_keys" "keys_with_label" {
+  with_selector = "foo=bar"
+}
+data "hcloud_ssh_keys" "empty_set" {
+  with_selector = "foo=baz"
+}
+`
+
+func TestAccHcloudDataSourceSSHKeys(t *testing.T) {
+	rInt := acctest.RandInt()
+	publicKeyMaterial, _, err := acctest.RandSSHKeyPair("hcloud-ds@ssh-acceptance-test")
+	if err != nil {
+		t.Fatalf("Cannot generate test SSH key pair: %s", err)
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccHcloudPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccHcloudCheckSSHKeysDataSourceConfig(rInt, publicKeyMaterial),
+			},
+			{
+				Config: dataSSHKeysConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.hcloud_ssh_keys.all_keys", "keys.0.name", fmt.Sprintf("sshkey-%d", rInt)),
+					resource.TestCheckResourceAttr(
+						"data.hcloud_ssh_keys.keys_with_label", "keys.0.name", fmt.Sprintf("sshkey-%d", rInt)),
+					resource.TestCheckResourceAttr(
+						"data.hcloud_ssh_keys.empty_set", "keys.#", "0"),
+				),
+			},
+		},
+	})
+}
+func testAccHcloudCheckSSHKeysDataSourceConfig(rInt int, key string) string {
+	return fmt.Sprintf(`
+resource "hcloud_ssh_key" "sshkeys_ds" {
+  name       = "sshkey-%d"
```

Comment:
Just for more difference to the other keys which are generated within the test i would name it something like `sshkey-ds-many-%d` just to have the difference....

Reply:
For consistency with the data.ssh_key test (which uses `sshkey-%d` for the name, how about `sshkeys-%d`? Pushing that up soon, let me know what you think

Reply:
Yep sounds good!